### PR TITLE
Add a response that mailserver failed to complete request.

### PR DESCRIPTION
**Pull request by mandrigin**
a part of the task of returning errors from mailserver instead of just being silent.
introduced a new type of possible response in this PR

### CODE REVIEW

**Comment by mandrigin** [comment]
@pilu agree

**Comment by mandrigin** [comment]
@pilu https://github.com/status-im/whisper/issues/5, feel free to comment/edit

File: whisperv6/whisper.go [hunk 1]
```diff
@@ -891,22 +901,36 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 
 				requestID = common.BytesToHash(payload[:common.HashLength])
 
-				if len(payload) >= common.HashLength*2 {
-					lastEnvelopeHash = common.BytesToHash(payload[common.HashLength : common.HashLength*2])
+				if isErrorPayload(payload) {
+					whisper.envelopeFeed.Send(EnvelopeEvent{
+						Hash:  requestID,
+						Event: EventMailServerRequestFailed,
+						Data:  string(payload[common.HashLength:]),
```
Comment (adambabik) [inline whisperv6/whisper.go]: Should we also get rid of the `ERROR=` prefix?

Comment (mandrigin) [reply whisperv6/whisper.go]: oh, right... tomorrow! :)

**Comment by pilu** [comment]
> @pilu #5, feel free to comment/edit

awesome thank you @mandrigin !

File: whisperv6/utils.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
+package whisperv6
```
Comment (adambabik) [inline whisperv6/utils.go]: I would extract mailserver related functions to `mailserver_response.go`.

Comment (mandrigin) [reply whisperv6/utils.go]: I can just rename the file, it is all about mailserver :)

**Comment by mandrigin** [comment]
@adambabik 100%, let's do it the right way... I will continue with that tomorrow.

**Comment by mandrigin** [comment]
@adambabik ah, right... probably we should keep this "serialization" as is then, because it mixes-and-matches well with another responses.

**Comment by mandrigin** [comment]
@adambabik addressed your feedback and squashed/signed commits.

**Comment by adambabik** [comment]
LGTM. Please merge and create a tag so that it can be used in status-go `Gopkg.toml` file. The latest can be visible here https://github.com/status-im/whisper/releases.

**Pull request merged**